plumbing/format/pktline/pktline_bench_test.go

```diff
@@ -0,0 +1,91 @@
+package pktline_test
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/go-git/go-git/v5/plumbing/format/pktline"
+)
+
+func BenchmarkReadPacket(b *testing.B) {
+	sections, err := sectionsExample(2, 4)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	cases := []struct {
+		name  string
+		input string
+	}{
+		{
+			name:  "empty",
+			input: "",
+		},
+		{
+			name:  "one message",
+			input: "000ahello\n",
+		},
+		{
+			name:  "two messages",
+			input: "000ahello\n000bworld!\n",
+		},
+		{
+			name:  "sections",
+			input: sections.String(),
+		},
+	}
+	for _, tc := range cases {
+		b.Run(tc.name, func(b *testing.B) {
+			r := strings.NewReader(tc.input)
+			for i := 0; i < b.N; i++ {
+				for {
+					_, _, err := pktline.ReadPacket(r)
+					if err != nil {
+						break
+					}
+				}
+			}
+		})
+	}
```

Comment:
I believe we want to reset the reader before each call:

```suggestion
	for _, tc := range cases {
		r := strings.NewReader("")
		b.Run(tc.name, func(b *testing.B) {
			for i := 0; i < b.N; i++ {
				r.Reset(tc.input)
				_, _, err := pktline.ReadPacket(r)
				if err != nil {
					break
				}
			}
		})
	}
```

This results on more sensible benchmarks - note the gap between an empty input and the rest:

```
goos: linux
goarch: amd64
pkg: github.com/go-git/go-git/v5/plumbing/format/pktline
cpu: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
BenchmarkReadPacket/empty-16         	1000000000	         0.0000005 ns/op	       0 B/op	       0 allocs/op
BenchmarkReadPacket/one_message-16   	12724945	        82.79 ns/op	      40 B/op	       3 allocs/op
BenchmarkReadPacket/two_messages-16  	14444020	        83.35 ns/op	      40 B/op	       3 allocs/op
BenchmarkReadPacket/sections-16      	13942911	        86.68 ns/op	      40 B/op	       3 allocs/op
```